packages/design-system-react-native/src/primitives/AvatarBase/AvatarBase.tsx

```diff
@@ -0,0 +1,47 @@
+/* eslint-disable @typescript-eslint/prefer-nullish-coalescing */
+import { useTailwind } from '@metamask/design-system-twrnc-preset';
+import React, { useMemo } from 'react';
+import { View } from 'react-native';
+
+import Text from '../../components/Text';
+import type { TextProps } from '../../components/Text/Text.types';
+import { DEFAULT_AVATARBASE_PROPS } from './AvatarBase.constants';
+import type { AvatarBaseProps } from './AvatarBase.types';
+import { generateAvatarBaseContainerClassNames } from './AvatarBase.utilities';
+
+const AvatarBase = ({
+  children,
+  size = DEFAULT_AVATARBASE_PROPS.size,
+  shape = DEFAULT_AVATARBASE_PROPS.shape,
+  fallbackText,
+  fallbackTextProps,
+  twClassName = '',
+  style,
+  ...props
+}: AvatarBaseProps) => {
+  const tw = useTailwind();
+  const twContainerClassNames = useMemo(() => {
+    return generateAvatarBaseContainerClassNames({
+      size,
+      shape,
+      twClassName,
+    });
+  }, [size, shape, twClassName]);
+
+  const finalFallbackTextProps: Omit<TextProps, 'children'> = {
+    ...DEFAULT_AVATARBASE_PROPS.fallbackTextProps,
+    ...fallbackTextProps,
+  };
+
+  return (
+    <View style={[tw`${twContainerClassNames}`, style]} {...props}>
+      {fallbackText ? (
+        <Text {...finalFallbackTextProps}>{fallbackText}</Text>
```

Comment:
I suggest setting default props at the prop definition level rather than handling defaults in the component body:

```suggestion
  fallbackTextProps = DEFAULT_AVATARBASE_PROPS.fallbackTextProps,
  twClassName = '',
  style,
  ...props
}: AvatarBaseProps) => {
  const tw = useTailwind();
  const twContainerClassNames = useMemo(() => {
    return generateAvatarBaseContainerClassNames({
      size,
      shape,
      twClassName,
    });
  }, [size, shape, twClassName]);

  return (
    <View style={[tw`${twContainerClassNames}`, style]} {...props}>
      {fallbackText ? (
        <Text {...fallbackTextProps}>{fallbackText}</Text>
```

This would:
1. Eliminate unnecessary object creation on each render
2. Follow React functional component best practices
3. Make the code more concise and readable
4. Maintain consistency with how other props are handled

This pattern is already used for size, shape, and twClassName props.